js.go

```diff
@@ -1019,52 +1019,80 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync
 		}
 
 		resp, err := js.nc.Request(js.apiSubj(ccSubj), j, js.opts.wait)
-		if err != nil {
+		if err == ErrTimeout {
```

Comment:
If we fix the behavior in the server will this change?

Reply:
Yes could be improved if the server sends a response such as `consumer already exists` at this point.

Reply:
Fixed by the server, so removed this code path.